pkg/tnf/reel/reel.go

```diff
@@ -254,15 +279,31 @@ func (r *Reel) wrapTestCommand(cmd string) string {
 // WrapTestCommand wraps cmd so that the output will end in an emulated terminal prompt.
 func WrapTestCommand(cmd string) string {
 	cmd = strings.TrimRight(cmd, "\n")
-	return fmt.Sprintf("%s && echo %s\n", cmd, EndOfTestSentinel)
+	wrappedCommand := fmt.Sprintf("%s ; echo %s %s$?\n", cmd, EndOfTestSentinel, ExitKeyword)
+	log.Tracef("Command sent: %s", wrappedCommand)
+	return wrappedCommand
 }
 
 // stripEmulatedPromptFromOutput will elide the emulated terminal prompt from the test output.
-func (r *Reel) stripEmulatedPromptFromOutput(output string) string {
-	if !r.disableTerminalPromptEmulation {
-		return strings.TrimRight(strings.TrimRight(output, endOfTestSentinelCutset), "\n")
+func (r *Reel) stripEmulatedPromptFromOutput(output string) (data string, status int) {
+	parsed := strings.Split(output, EndOfTestSentinel)
+	if !r.disableTerminalPromptEmulation && len(parsed) > 1 {
```

Comment:
= 2 instead? that's the only case supported by this branch, isn't it?

Reply:
also, can we still support empty data, i.e. len =1? just use parsed[0] for status parsing and set data to ""?

Reply:
Fixed ==2
The case with empty data is covered also by ==2. parsed[0] will be "" in this case

Reply:
cool, i thought this was the reason why empty patterns were no longer supported. I see you have added them back.